Assets/Code/Scripts/PlayerManagement/CameraManager.cs

```diff
@@ -31,21 +38,30 @@ void Start()
         _realPlayerMovement.ActivatePlayer(true);
         _realMouseLook.ActivateMouseLook(true);
         
-        _noClipPlayerCamera.SetActive(false);
-        _noClipMouseLook.ActivateMouseLook(false);
+        _noclipCamera.SetActive(false);
+        _noclipMovement.ActivatePlayer(false);
+        _noclipMouseLook.ActivateMouseLook(false);
         
         _activeRealPlayer = true;
     }
     
     public void SwitchCamera()
     {
+        //Activate/disactivate the realPlayer and his camera
         _realPlayerCamera.SetActive(!_activeRealPlayer);
         _realPlayerMovement.ActivatePlayer(!_activeRealPlayer);
         _realMouseLook.ActivateMouseLook(!_activeRealPlayer);
-            
-        _noClipPlayerCamera.SetActive(_activeRealPlayer);
-        _noClipMouseLook.ActivateMouseLook(_activeRealPlayer);
-            
+        
+        if (_activeRealPlayer) //When the switch from reality mode to noclip mode happened
+        {
+            _noclipMovement.SetPositionAndRotation(_realPlayer.transform.position, _realPlayer.transform.rotation, _realPlayerCamera.transform.rotation); //Set the noclip position in the realBody position
+        }
+        
+        //Activate/disactivate the noclipPlayer and his camera
+        _noclipCamera.SetActive(_activeRealPlayer);
+        _noclipMovement.ActivatePlayer(_activeRealPlayer);
```

Comment:
This function sets the `_noclipMovement._currentPlayer` to True or False when `_activeRealPlayer` is True or False. So, the two variables have the same value. But aren't they supposed to have opposite values? Or am I just confused?